app/controllers/order_items_controller.rb

```diff
@@ -0,0 +1,22 @@
+class OrderItemsController < ApplicationController
+  before_action :require_login
+  before_action :find_orderitem, only: [:edit, :update]
```

Comment:
Good use of the before_action here! I especially like this in a short controller as it doesn't get lost-- I have had issues with before_actions (dependent ones in another controller) in my internship that made it really hard to follow/debug-- I thought this article was an interesting take: http://blog.thefrontiergroup.com.au/2014/02/before_action_an_anti_pattern/ (good to read with a grain of salt and generally follow the style/convention of Ada class/your internship/job)